var/spack/repos/builtin/packages/damask-mesh/package.py

```diff
@@ -17,22 +17,19 @@ class DamaskMesh(CMakePackage):
 
     depends_on('pkgconfig', type='build')
     depends_on('cmake@3.10:', type='build')
+    depends_on('petsc+mpi+hdf5')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.15', when='@3.0.0-alpha4')
     depends_on('petsc@3.14.0:3.14,3.15.1:3.16', when='@3.0.0-alpha5')
-    depends_on('hdf5+fortran')
+    depends_on('hdf5@1.10:+mpi+fortran')
```

Comment:
The other aspect of this change is that now MPI is *required* (transitively) to build damask-mesh due to this and the new `petsc` requirement. Is that intended? Same with `damask-grid`. Do you mean to do something like:
```python
conflicts('hdf5+mpi', when='^hdf5@:1.10')
```
for the MPI error (previously your requirement for 1.12+), and
```python
depends_on('hdf5+mpi', when='^petsc +mpi')
```
to specify that it requires parallel HDF5 when built against parallel petsc?

Reply:
PETSc with MPI support is required for both solvers (mesh and grid), also for older versions of DAMASK. FFTW and HDF5 also need MPI support.

> Parallel IO is therefore disabled for versions < 1.12. Using 1.10 should be considered as a fallback option.

I have to correct myself: For older versions of HDF5 (<1.12), compression during parallel IO is disabled, not MPI-parallel IO in general. But disabling compression is a runtime operation. 

As far as I see, `petsc+mpi` requires `hdf5+mpi`,
https://github.com/spack/spack/blob/897616f073a4da58fd48213db44a0e2cdcd83f1c/var/spack/repos/builtin/packages/petsc/package.py#L98-L99

So `depends_on('hdf5@1.10:+mpi+fortran')` and `depends_on('hdf5@1.10:+fortran')` should result in the same setup. The first option is just more explicit.